dashboard/src/storage/StorageList.js

```diff
@@ -0,0 +1,208 @@
+            stateColor={item.state_color}
+            storageClass={item.storage_class}
+            storageClassIsDefault={item.storage_class_is_default}
+            deleteCommand={createDeleteCommand(item.name)}
+            describeCommand={createDescribeCommand(item.name)}
+          />
+        ) : <p>No items</p>
+      }
+    </Table.Body>
+  </Table>
+);
+
+class RowComponent extends Component {
+  state = {expanded: true};
+
+  onToggleExpand = () => { this.setState({expanded: !this.state.expanded});}
+
+  render() {
+    return [<RowView 
+      key="datarow"
+      name={this.props.name}
+      localPaths={this.props.localPaths}
+      stateColor={this.props.stateColor}
+      storageClass={this.props.storageClass}
+      storageClassIsDefault={this.props.storageClassIsDefault}
+      deleteCommand={this.props.deleteCommand}
+      describeCommand={this.props.describeCommand}
+      toggleExpand={this.onToggleExpand}
+      expanded={this.state.expanded}
+    />,
+    this.state.expanded && <VolumesRowView
+      key="volrow"
+      name={this.props.name}
+      expanded={this.state.expanded}
+      toggleExpand={this.onToggleExpand}
+    />
+    ];
+  }
+}
+
+const EmptyView = () => (<div>No local storage resources</div>);
+
+function createDeleteCommand(name) {
+  return `kubectl delete ArangoLocalStorage ${name}`;
+}
+
+function createDescribeCommand(name) {
+  return `kubectl describe ArangoLocalStorage ${name}`;
+}
+
+function getStateColorDescription(stateColor) {
+  switch (stateColor) {
+    case "green":
+      return "Everything is running smooth.";
+    case "yellow":
+      return "There is some activity going on, but local storage is available.";
+    case "orange":
+      return "There is some activity going on, local storage may be/become unavailable. You should pay attention now!";
+    case "red":
+      return "The local storage is in a bad state and manual intervention is likely needed.";
+    default:
+      return "State is not known.";
+  }
+}
+
+class StorageList extends Component {
+  state = {
+    items: undefined,
+    error: undefined,
+    loading: true
+  };
+
+  componentDidMount() {
+    this.reloadStorages();
+  }
+
+  reloadStorages = async() => {
+    try {
+      this.setState({
+        loading: true
+      });
+      const result = await api.get('/api/storage');
+      this.setState({
+        items: result.storages,
+        loading: false,
+        error: undefined
+      });
+    } catch (e) {
+      this.setState({
+        error: e.message,
+        loading: false
+      });
+      if (IsUnauthorized(e)) {
+        this.props.doLogout();
+        return;
+      }
+    }
+    this.props.setTimeout(this.reloadStorages, 5000);
+  }
+
+  render() {
+    const items = this.state.items;
+    if (!items) {
+      return (<Loading />);
+    }
+    if (items.length === 0) {
+      return (<EmptyView />);
+    }
+    return (<ListView items={items} loading={this.state.loading} />);
+  }
+}
+
```

Comment:
This file is getting rather large. I'd split it out to a folder. I generally aim to have one component per folder but YMMV.

Reply:
once all the basic are in, I'll go over all of it for issues like this.